python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -0,0 +1,923 @@
+Changing the value of a variable is as simple as editing the value in the assignment statement and re-running the code; alternatively, you can write a new assignment statement assigning the updated value to the variable.
+</div>
+***
+
+## Lists
+
+**Lists** in Python are objects that contain a collection of multiple items. They are surrounded by square brackets [ ] and the items in the list are separated by commas:
+
+@sage
+<div class="python_link">
+<lia-keep>
+<script type="text/x-sage">
+
+produce = ["tomato", "squash", "apple", "cucumber", "peach", "spinach"]
+print(produce)
+
+</script>
+</lia-keep>
+</div>
+
+There are a few important characteristics of lists:
+
+* Lists can have duplicate values. An example might be the list `[3, 1, 4, 1, 5, 9]`, where the number 1 appears twice.
+
+* List items don't all have to be of the same type (you can have a mix of strings and numbers, for example, or even a list of lists). `[3.14, "pi", False, ["number", "string", "boolean"]]` is an example of a list that contains a float, a string, a boolean value, and a list as its elements.
+
+* Lists are **ordered**: List items remain in the position that they are entered into the list, and you can access any item by its position, or **index**.
+
+  <div class = "important">
+  An **index** in Python refers to a value's position in a sequence. The sequence can be a collection object like a list or even a string (which you can think of as a sequence of characters). The important thing to remember is that in Python, **indices start at 0**. This means that the index of the first item in a sequence is `0`! This is not always the case in other programming languages.
+
+  To access an item by its index, you use **subsetting notation**, `sequence[index]`. To get the first letter of the string "Hello", for example, you would use the code `"Hello"[0]`.
+  </div>
+
+* Lists are **changeable**; you can add, remove, and edit items in a list. There are a few ways to do this:
+
+    * `.pop()` removes a specific element by **index**).
+
+     @sage
+     <div class="python_link">
+     <lia-keep>
+     <script type="text/x-sage">
+
+     print(produce.pop(0))
+
+     </script>
+     </lia-keep>
+     </div>
+
+   * `.remove()` removes a specific element by **value**.
+
+    @sage
+    <div class="python_link">
+    <lia-keep>
+    <script type="text/x-sage">
+
+    produce.remove("apple")
+print(produce)
+
+    </script>
+    </lia-keep>
+    </div>
+
+  * `del` deletes objects, and can be used as another way to remove elements of a list by index. The syntax will be a little different, because it is a keyword, not a list method. It is useful because it allows you to delete list elements in a slice. Slicing uses subsetting notation. the first index listed is **inclusive**, or included in the slice; this is followed by a colon; the second index listed is **exclusive**, or excluded from the slice; finally, these numbers are surrounded by square brackets.
```

Comment:
```suggestion
  * `del` deletes objects, and can be used as another way to remove elements of a list by index. The syntax will be a little different, because it is a keyword, not a list method. It is useful because it allows you to delete list elements in a **slice**. Slicing uses subsetting notation to indicate where the slice begins and ends. The first index listed is **inclusive**, or included in the slice; this is followed by a colon; the second index listed is **exclusive**, or excluded from the slice; finally, these numbers are surrounded by square brackets.
```